src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,131 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> tailNode;
+    private Node<T> headNode;
+    private Node<T> currentNode;
+    private int currentNodeIndex;
+
+    private Node<T> newNode;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node newNode = new Node(null, value, null);
+            headNode = newNode;
+            tailNode = newNode;
+        } else {
+            Node newNode = new Node(tailNode, value, null);
+            tailNode.next = newNode;
+            tailNode = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        goToNodeByIndex(index);
+        if (index == 0) {
+            newNode = new Node(null, value, currentNode);
+            size++;
+            headNode = newNode;
+            return;
+        }
+        newNode = new Node(currentNode.prev, value, currentNode);
+        size++;
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void addAll(List<T> myList) {
+        if (!myList.isEmpty()) {
```

Comment:
Why do we need this check?

Reply:
when we have empty list, than we do nothing.
we can delete it.